### PR TITLE
PROD-699 #3124 - Fix: Preview link is not working when no any keystroke or only using right-click to paste the URL

### LINKED ISSUE #3124: Preview link is not working when no any keystroke or only using right-click to paste the URL
**Describe the bug**
Preview link is not working when no any keystroke or only using right-click to paste the URL. Screen recording: https://cutt.ly/aRKILgC

**To Reproduce**
1. Copy any link from YouTube/Vimeo
2. Go to the News Feed and paste the URL using your mouse, do not add any text or use any keyboard shortcuts
3. click post and notice the issue

**Environment**
demo, dev, local

**Support ticket links**
https://secure.helpscout.net/conversation/1665683646/169485?folderId=3709082
**Jira issue** : [PROD-699]

[PROD-699]: https://buddyboss.atlassian.net/browse/PROD-699?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### PR DESCRIPTION
**Pull request by milanhirapra**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #3124 .

### How to test the changes in this Pull Request:

1. Copy any link from YouTube/Vimeo
2. Go to the News Feed and paste the URL using your mouse, do not add any text or use any keyboard shortcuts
3. click post and notice the preview is working fine

### Proof Screenshots or Video
https://vimeo.com/669320512/2b9829e112

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Review by chetansatasiya — APPROVED** [review]
LGTM 🚀

**Comment by aprilgeller** [comment]
Retested on QA ongoing branch. 
Status: Not yet Fixed

Proof Video recording: https://www.loom.com/share/3117359b2ad94213a58a7a4a32090a1a

File: src/bp-activity/bp-activity-filters.php [hunk 1]
```diff
@@ -160,6 +160,9 @@
 // Filter check content empty or not for the media, document and GIF data.
 add_filter( 'bb_is_activity_content_empty', 'bb_check_is_activity_content_empty' );
 
+// Filter check the single embed URL wrap with "P" tag or not.
+add_filter( 'bp_activity_content_before_save', 'bb_activity_content_has_paragraph_tag' );
```
Comment (KartikSuthar) [inline src/bp-activity/bp-activity-filters.php:164]: @milanhirapra please remove this filter its not needed.

File: src/bp-activity/bp-activity-filters.php [hunk 2]
```diff
@@ -2997,3 +3000,28 @@ function bb_activity_delete_link_review_attachment( $activities ) {
...
+
+	preg_match( '/(https?:\/\/[^\s<>"]+)/i', $content, $content_url );
+	preg_match( '(<p(>|\s+[^>]*>).*?<\/p>)', $content, $content_tag );
+
+	if ( ! empty( $content_url ) && empty( $content_tag ) ) {
+		$content = sprintf( '<p>%s</p>', $content );
+	}
+
+	return $content;
+}
```
Comment (KartikSuthar) [inline src/bp-activity/bp-activity-filters.php:3027]: Remove this code from here.
add it on: https://github.com/buddyboss/buddyboss-platform/blob/release/src/bp-core/classes/class-bp-embed.php#L299

```
if ( ! empty( $embed_code ) ) {

	if ( ! empty( $content ) ) {
		preg_match( '/(https?:\/\/[^\s<>"]+)/i', $content, $content_url );
		preg_match( '(<p(>|\s+[^>]*>).*?<\/p>)', $content, $content_tag );

		if ( ! empty( $content_url ) && empty( $content_tag ) ) {
			$content = sprintf( '<p>%s</p>', $content );
		}
	}

	$content .= $embed_code;
	$link_embed = true;
}
```

**Review by KartikSuthar — CHANGES_REQUESTED** [review]

**Review by KartikSuthar — APPROVED** [review]

**Pull request merged**